Graphics/GraphicsEngineD3D12/include/PipelineStateD3D12Impl.hpp

```diff
@@ -129,7 +129,7 @@ class PipelineStateD3D12Impl final : public PipelineStateBase<IPipelineStateD3D1
     ShaderResourceCacheD3D12*   m_pStaticResourceCaches  = nullptr;
     ShaderVariableManagerD3D12* m_pStaticVarManagers     = nullptr;
     // Resource layout index in m_ResourceLayouts[] array for every shader stage
-    Int8 m_ResourceLayoutIndex[6] = {-1, -1, -1, -1, -1, -1};
+    std::array<Int8, NUM_SHADER_TYPES> m_ResourceLayoutIndex;
```

Comment:
You removed default initialization here, but did not add it to the constructor. This could've resulted in a very hard-to-find issues.